docs/develop/initialize-add-in.md

```diff
@@ -0,0 +1,147 @@
+
+- Prompting the user to select some cells in Excel, and then inserting a chart initialized with those selected values.
+
+- Establish bindings.
+
+- Use the Office dialog API to prompt the user for default add-in settings values.
+
+However, an Office Add-in cannot successfully call any Office JavaScript APIs until the library has been loaded. This article describes the two ways your code can ensure that the library has been loaded:
+
+- Initialize with `Office.onReady()`.
+- Initialize with `Office.initialize`.
+
+> [!TIP]
+> We recommend that you use `Office.onReady()` instead of `Office.initialize`. Although `Office.initialize` is still supported, using `Office.onReady()` provides more flexibility. You can assign only one handler to `Office.initialize` and it's called only once by the Office infrastructure, but you can call `Office.onReady()` in different places in your code and use different callbacks.
+> 
+> For information about the differences in these techniques, see [Major differences between Office.initialize and Office.onReady()](#major-differences-between-officeinitialize-and-officeonready).
+
+For more details about the sequence of events when an add-in is initialized, see [Loading the DOM and runtime environment](loading-the-dom-and-runtime-environment.md).
+
+## Initialize with Office.onReady()
+
+`Office.onReady()` is an asynchronous method that returns a Promise object while it checks to see if the Office.js library is loaded. When the library is loaded, it resolves the Promise as an object that specifies the Office host application with an `Office.HostType` enum value (`Excel`, `Word`, etc.) and the platform with an `Office.PlatformType` enum value (`PC`, `Mac`, `OfficeOnline`, etc.). The Promise resolves immediately if the library is already loaded when `Office.onReady()` is called.
+
+One way to call `Office.onReady()` is to pass it a callback method. Here's an example:
+
+```js
+Office.onReady(function(info) {
+    if (info.host === Office.HostType.Excel) {
+        // Do Excel-specific initialization (for example, make add-in task pane's
+        // appearance compatible with Excel "green").
+    }
+    if (info.platform === Office.PlatformType.PC) {
+        // Make minor layout changes in the task pane.
+    }
+    console.log(`Office.js is now ready in ${info.host} on ${info.platform}`);
+});
+```
+
+Alternatively, you can chain a `then()` method to the call of `Office.onReady()`, instead of passing a callback. For example, the following code checks to see that the user's version of Excel supports all the APIs that the add-in might call.
+
+```js
+Office.onReady()
+    .then(function() {
+        if (!Office.context.requirements.isSetSupported('ExcelApi', '1.7')) {
+            console.log("Sorry, this add-in only works with newer versions of Excel.");
+        }
+    });
+```
+
+Here is the same example using the `async` and `await` keywords in TypeScript:
+
+```typescript
+(async () => {
```

Comment:
For my own learning, why is this async function wrapped in parentheses?

Reply:
I have no idea why it's wrapped in parentheses. The contents of this article was copied/moved as-is in its entirety from the old "Understanding the Office JavaScript API" article, which was written by someone else (Ricky?).

Reply:
@AlexJerabek @kbrandl It's an [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE), which insures that it runs as soon as the file that's in loads. @Zlatkovsky  was the subject matter expert. He may remember why an IIFE is used for TypeScript and not JavaScript.